### PR TITLE
[example] Update Llama Inference example

### PR DESCRIPTION
**Pull request by yuanheng-zhao**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

1. Update Llama3 Config and inference settings in example benchmark.
2. Add llama generation demo script (Llama3)
3. Add a clean-up version of Llama3 benchmark
4. fix init rope in llama policy

image

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/inference/benchmark_llama.py [hunk 1]
```diff
@@ -117,6 +133,8 @@ def benchmark_inference(args):
                 max_output_len=args.output_len,
                 prefill_ratio=1.2,
                 block_size=32,
+                tp_size=args.tp_size,
+                pp_size=args.pp_size,
```
Comment (yuehuayingxueluo) [inline examples/inference/benchmark_llama.py:137]: Do we need pp at present?

File: examples/inference/benchmark_llama.py [hunk 2]
```diff
@@ -51,6 +51,22 @@
...
     ),
+    "llama3-8b": transformers.LlamaConfig(
+        hidden_size=4096,
+        intermediate_size=14336,
+        num_attention_heads=32,
+        num_hidden_layers=32,
+        num_key_value_heads=8,
+        max_position_embeddings=8192,
+    ),
+    "llama3-70b": transformers.LlamaConfig(
```
Comment (yuehuayingxueluo) [inline examples/inference/benchmark_llama.py:62]: Can our current framework run llama3 70B?

Comment (yuanheng-zhao) [reply examples/inference/benchmark_llama.py:62]: Of course

File: examples/inference/benchmark_llama.py (hunk 1 above)
Comment (yuanheng-zhao) [reply examples/inference/benchmark_llama.py:137]: No need for now. Just keep the original configs/args.

Comment (yuanheng-zhao) [reply examples/inference/benchmark_llama.py]: Unused args removed.

**Review by yuehuayingxueluo — APPROVED** [review]

**Comment by yuanheng-zhao** [comment]
A clean-up version of Llama benchmark is added (`examples/inference/benchmark_llama3.py`) so that users won't need to install vLLM to run benchmark (the old-comparison one). The clean-up version remove unused args and invalid statements, only focusing on benchmark Colossal-Inference Engine.

**Comment by yuanheng-zhao** [comment]
CI - Test Example on PR will be fixed in another PR fixing Inference CI issues.

**Pull request merged**